encoder/vaapiencoder_h264.cpp

```diff
@@ -1142,10 +1326,17 @@ bool  VaapiEncoderH264::pictureReferenceListSet (
 
     for (i = 0; i < m_refList.size(); i++) {
         assert(picture->m_poc != m_refList[i]->m_poc);
-        if (picture->m_poc > m_refList[i]->m_poc)
+        if (picture->m_poc > m_refList[i]->m_poc
```

Comment:
Let me explain my concern in details:
1.	Original code only have two branches if else, this code have three branches if, else if, else. Could we change it to two branches?
2.	If yes, maybe we can change code like this, it maybe more readable
```
if (picture->m_temporalId >= m_refList[i]->m_temporalId) {
	m_refList[i]->m_diffPicNumMinus1 = picture->m_frameNum - m_refList[i]->m_frameNum - 1;
}
if (picture->m_poc > m_refList[i]->m_poc)
	m_refList0.push_back(m_refList[i]);/* set forward reflist: descending order */
else
	m_refList1.push_front(m_refList[i]);/* set backward reflist: ascending order */
```